### ISSUE TITLE
What is the proper way to fully switch a homeserver from federation to standalone operation and stop attempts to contact outside homeservers?

### ISSUE THREAD

**Issue by derzahla** [matrix-org/synapse#12481]
I have seen many issues touching on this topic but no conclusive answers here or in documentation. I manage a homeserver that has been operating standalone for many months now but it still peppers the logs trying to reach out the dozens of external homeservers.

How can I make this stop?

I have already tried the following which were suggested in other reponses:

1. removed the federation listener from homeserver.yml
2. set the federation whitelist to " "
3. purged all old rooms that were left over from federation
4. manually cleared all rows in the "destinations" and "destination_room" database tables

 what exactly needs to be done to stop all federation activity and remove all record of previously federated servers?

**Comment by DMRobertson** [comment]
>  but it still peppers the logs trying to reach out the dozens of external homeservers.

Can you show us an example log line?

**Comment by richvdh** [comment]
related: https://github.com/matrix-org/synapse/issues/6401

**Comment by dklimpel** [comment]
Related: #3765

**Comment by derzahla** [comment]
Here is an example, but there are many more.   Do I need to clear certain tables in the DB?  Or is there a way to just shut down the federation process? 

`2022-04-17 12:42:50,453 - synapse.metrics.background_process_metrics - 244 - ERROR - process_event_queue_for_federation-4- Background process 'process_event_queue_for_federation' threw an exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/synapse/metrics/background_process_metrics.py", line 242, in run
    return await func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/synapse/federation/sender/__init__.py", line 342, in _process_event_queue_loop
    next_token, events = await self.store.get_all_new_events_stream(
  File "/usr/local/lib/python3.8/site-packages/synapse/storage/databases/main/stream.py", line 999, in get_all_new_events_stream
    events = await self.get_events_as_list(event_ids)
  File "/usr/local/lib/python3.8/site-packages/synapse/storage/databases/main/events_worker.py", line 463, in get_events_as_list
    event_entry_map = await self._get_events_from_cache_or_db(
  File "/usr/local/lib/python3.8/site-packages/synapse/storage/databases/main/events_worker.py", line 654, in _get_events_from_cache_or_db
    raise e
  File "/usr/local/lib/python3.8/site-packages/synapse/storage/databases/main/events_worker.py", line 646, in _get_events_from_cache_or_db
    missing_events = await self._get_events_from_db(
  File "/usr/local/lib/python3.8/site-packages/synapse/storage/databases/main/events_worker.py", line 1008, in _get_events_from_db
    raise Exception(
Exception: Room !LjiiSHrYKkIbVXUnCy:glowers.club for event $bJUGrXKm58c_xwD4UZW04Z2nPk8vY97zR63F-G_aTyw is unknown
`

**Comment by hparadiz** [comment]
I've been having the same problem recently

**Comment by reivilibre** [comment]
> Exception: Room !LjiiSHrYKkIbVXUnCy:glowers.club for event $bJUGrXKm58c_xwD4UZW04Z2nPk8vY97zR63F-G_aTyw is unknown

How did you purge the old rooms? Using a query in the database or using the admin API?

**Comment by derzahla** [comment]
Sorry for the late reply.  My reason for opening this was the noise which I thought may be related to major problems my homeserver was having.  I ended up fixing it by rolling back to an old backup so the root problem was unrelated to this.  I purged the rooms through the API and I did already clear the two destinations tables per the advice I found on another case. It 
would be great to have a document outlining how to completely remove all federation hooks.
